### PR TITLE
feat(nav): implement navigation states for default answers

### PR DESCRIPTION
**Pull request by anehx**
Navigation items which have fields that have only default answers will be greyed out to indicate that the user didn't do anything there but will still show the state (check, times or question mark) to indicate the state of the data in there:

Screenshot from 2020-11-25 14-47-23

### CODE REVIEW

File: addon/lib/field.js [hunk 1]
```diff
@@ -187,6 +187,39 @@ export default Base.extend({
...
+    "question.{isTable,defaultValue}",
+    function () {
+      if (!this.value || !this.question.defaultValue) return false;
+
+      const value = this.question.isTable
+        ? this.value.map((doc) => doc.flatAnswerMap)
+        : this.value;
+
+      return (
+        JSON.stringify(value) === JSON.stringify(this.question.defaultValue)
```
Comment (fkm-adfinis) [inline addon/lib/field.js:207]: This worried me a bit as the order of object properties is not guaranteed. But from what I've learned they should respect the order they were set in.

So, my question becomes: Is the order of `this.fields` dependent on configuration or the order in which they were filled in by the user? If it is by configuration, this comparison should not be a problem.

Comment (anehx) [reply addon/lib/field.js:207]: Very good point. The order of the fields is based on configuration, however the order of the default answer is not.. I'll have to find another way of comparing those. Do you have experience with https://lodash.com/docs/4.17.15#isEqual ?

Comment (fkm-adfinis) [reply addon/lib/field.js:207]: I remember using it a lot with React to check if the properties had changed. *I don't remember having any issues with it.* But, that was a long time ago and maybe I suppressed a bad memory including isEqual :smile: 

But yeah. I agree that this sounds like a good idea!

Comment (anehx) [reply addon/lib/field.js:207]: Tested it and it works. I also added a test, will push soon. Good catch!

Comment (fkm-adfinis) [reply addon/lib/field.js]: :+1:

**Review by fkm-adfinis — APPROVED** [review]

**Comment by kaldras** [comment]
Very nice change! I believe this will bring more clarity for the user to what work should or can be done and the overall state of the form! :+1:

**Review by kaldras — APPROVED** [review]

**Pull request merged**